deeplake/core/vectorstore/vector_search/utils.py

```diff
@@ -19,6 +21,19 @@ def parse_tensor_return(tensor):
     return tensor.data(aslist=True)["value"]
 
 
+def parse_exec_option(dataset, exec_option):
+    if exec_option == "auto":
+        if isinstance(dataset, DeepLakeCloudDataset):
+            if "vector_db/" in dataset.base_storage.path:
+                return "tensor_db"
+            else:
+                return "compute_engine"
```

Comment:
Not sure why in case of `DeepLakeCloudDataset` by default it should be either `tensor_db` or `compute_engine`. Shouldn't by default it be `tensor_db` or `python`? Or is it our final decision to do it like that?

What about `gcs`, `s3`, `azure`, do we want for all of them by default use `compute_engine`?  If so, we will need to do a quick follow up PR to LangChain to explicitly use python be default, as users might start getting problems

Reply:
Will explain on a call.

Reply:
Regarding the first comment, I'll explain that on our call. 

Regarding S3, GCS, etc, those dataset have to have a cloud path to be able to access compute_engine. Therefore, if they have a Deep Lake path, they'll use compute engine. If they are not connected to Deep Lake, they use python.